### PR TITLE
prepend plugins instead of append when defined through CLI

### LINKED ISSUE #4260: Order of precedence between command line argument and config files is inconsistent
**Do you want to request a *feature* or report a *bug*?**

I believe it's a bug.

**If the current behavior is a bug, please provide the steps to reproduce.**

I have 2 very simple test cases, sharing the same `package.json` (relevant only to use the latest webpack version) and the same `app.js`. Only differs the webpack config and the CLI arguments:
* https://gist.github.com/tbassetto/048524ef16a1a30ca293ad1c9b7e69bd
* https://gist.github.com/tbassetto/4a4aefdae71333519bdb62debcdf8379

In one example, the CLI has higher priority then the config file, in the other example it's the opposite.

**What is the expected behavior?**

Consistency (not sure which one is better).

**If this is a feature request, what is motivation or use case for changing the behavior?**

N/A

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

Latest macOS Sierra, webpack 2.2.1, Node.js 6.9.5 (latest LTS).

### PR DESCRIPTION
**Pull request by esbenp**
closes #4260

**What kind of change does this PR introduce?**

Gives plugins presedence over webpack config file when defined through the CLI

**Did you add tests for your changes?**

Unsure how to test this, but feel free til suggest. Will add if relevant

**If relevant, link to documentation update:**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

**Comment by sokra** [comment]
For testing see `test/binCases`. Best write a test having DefinePlugin `TEST: "fail"` in webpack.config.js and `--define TEST=ok` in CLI and `require("./" + TEST + ".js")` in entry.js. You can test the exit code for success.

**Comment by esbenp** [comment]
Creative :-) added in b9c1f9b

**Comment by sokra** [comment]
```
webpack/webpack/test/binCases/configFile/plugins-presedence/test.js
  7:30  error  Strings must use doublequote  quotes
  7:38  error  Missing semicolon             semi
```

File: bin/convert-argv.js [hunk 1]
```diff
@@ -330,7 +334,7 @@ module.exports = function(yargs, argv, convertOptions) {
 		}, function() {
 			ensureArray(options, "plugins");
```
Comment (sokra) [inline bin/convert-argv.js]: Could you move the `ensureArray(options, "plugin")` into the addPlugin method to cleanup the code a bit.

**Comment by esbenp** [comment]
Affirmative

**Comment by sokra** [comment]
Thanks

**Pull request merged**